frontend/js/Sidebar.js

```diff
@@ -0,0 +1,76 @@
+import { register, login, logout } from "./ApiCalls.js";
+
+//collapse sidebar on click
+const collapseButton = document.getElementById("collapseButton");
+const sidebar = document.getElementById("sidebar");
+const content = document.getElementById("content-container");
+
+collapseButton.addEventListener("click", toggleSidebarCollapse);
+
+function toggleSidebarCollapse() {
+  sidebar.classList.toggle("collapsed");
+  content.classList.toggle("collapsed");
+  var sectionNames = document.querySelectorAll(".section-name");
+  sectionNames.forEach(function (name) {
+    name.classList.toggle("section-name-collapsed");
+  });
+}
+
+//navbar active button switcher
+var navlist = document.getElementById("navList");
+navlist.addEventListener("click", (e) => {
+  let target = e.target.closest("a");
+  document.querySelector("a.active").classList.remove("active");
+  target.classList.add("active");
+});
+
+//login button fetch launcher
+var loginForm = document.getElementById("loginForm");
+loginForm.addEventListener("submit", (event) => login(event, loginForm));
+
+//logout button fetch launcher
+var logoutButton = document.getElementById("logoutButton");
+logoutButton.addEventListener("click", (e) => logout());
+
+//register button fetch launcher
+var registerForm = document.getElementById("registrationForm");
+registerForm.addEventListener("submit", (event) =>
+  register(event, registerForm)
+);
+
+//check password on register input
+var password1 = document.getElementById("passwordRegister");
+password1.addEventListener("input", (event) => checkPassword());
+
+var password2 = document.getElementById("passwordagain");
+password2.addEventListener("input", (event) => checkPassword());
+
+function checkPassword() {
+  var registerPassLength = document.getElementById("registerPassLength");
+  var registerPassLowercase = document.getElementById("registerPassLowercase");
+  var registerPassUppercase = document.getElementById("registerPassUppercase");
+  var registerPassNumbers = document.getElementById("registerPassNumbers");
+  var registerPassSpecial = document.getElementById("registerPassSpecial");
+
+  var helpTextAgain = document.getElementById("passwordagainHelpBlock");
+
+  if (/^.{8,20}$/.test(password1.value))
```

Comment:
You can create function for test: 
`function testPassword(critere, regex) {
 if (regex.test(password1.value))
    critere.style.color = "green";
  else critere.style.color = "red";
}`

Reply:
Done!